encoder/generator.go

```diff
@@ -478,14 +478,36 @@ func (g *generator) genInterface(name string, v cue.Value) {
 		d, ok := fields.Value().Default()
 		// [...number] results in [], which is not desired
 		// TODO: There must be a better way to handle this
-		if ok && d.IncompleteKind() != cue.ListKind {
-			dStr, err := tsprintField(d)
-			g.addErr(err)
-			kv.Default = dStr
-			if _, r := d.Reference(); len(r) > 0 {
-				kv.Default = strcase.ToLowerCamel(kv.Default + "Default")
+		if ok {
```

Comment:
this logic works now, it is better to move the logics into tsprintField, it makes more sense.